set.c

```diff
@@ -192,7 +191,7 @@ static const rb_data_type_t set_data_type = {
         .dmark = set_mark,
         .dfree = set_free,
         .dsize = set_size,
-        .dcompact = set_compact,
+        .dcompact = set_gc_compact,
```

Comment:
```suggestion
        .dcompact = set_update_references,
```

Is a convention used for a few other types.